src/project/iaudacityproject.h

```diff
@@ -39,7 +39,8 @@ class IAudacityProject
     virtual void setNeedAutoSave(bool val) = 0;
     virtual muse::async::Notification needSaveChanged() const { return muse::async::Notification(); }
     virtual muse::Ret save(const muse::io::path_t& path = muse::io::path_t(), SaveMode saveMode = SaveMode::Save) = 0;
-    virtual muse::async::Notification captureThumbnailRequested() const = 0;
+    virtual muse::async::Channel<std::string> captureThumbnailRequested() const = 0;
+    virtual void onThumbnailCreated(bool success) = 0;
```

Comment:
I suggest removing these methods from the project. In ProjectPropertiesModel, use the creator directly.
There is such logic here, the public interface for creating a thumb is the IThumbnailCreator. The ProjectPropertiesModel model is its implementation...
In the future, we can change the implementation, for example, to create Thumbnail, use not the current Qml scene grab, but independent drawing or something like that.

Reply:
Yes! I have refactor it.